File: src/main/java/com/elchologamer/userlogin/command/sub/SetCommand.java
package com.elchologamer.userlogin.command.sub;

import com.elchologamer.userlogin.ULPlayer;
import com.elchologamer.userlogin.UserLogin;
import com.elchologamer.userlogin.command.base.SubCommand;
import com.elchologamer.userlogin.util.QuickMap;
import org.bukkit.Location;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.ArrayList;
import java.util.List;

public class SetCommand extends SubCommand {

    private final UserLogin plugin = UserLogin.getPlugin();

    public SetCommand() {
        super("set", true, "ul.set");
    }

    @Override
    public boolean run(CommandSender sender, String[] args) {
        if (args.length == 0) return false;

        String type = args[0].toLowerCase();
        if (!type.equals("spawn") && !type.equals("login") && !type.equals("prelogin") && !type.equals("postlogin"))
            return false;

        if (type.equals("prelogin")) {
            type = "login";
        } else if (type.equals("postlogin")) {
            type = "spawn";
        }

        ULPlayer ulPlayer = ULPlayer.get((Player) sender);
        Player player = ulPlayer.getPlayer();

        // Save location
        Location loc = player.getLocation();
        plugin.getLocations().saveLocation(type, loc);

        // Send message
        ulPlayer.sendMessage(
                "commands.set",
                new QuickMap<>("type", (Object) args[0].toLowerCase())
                        .set("x", (int) loc.getX())
                        .set("y", (int) loc.getY())
                        .set("z", (int) loc.getZ())
                        .set("yaw", (int) loc.getYaw())
                        .set("pitch", (int) loc.getPitch())
                        .set("world", loc.getWorld().getName())
        );

        return true;
    }

    @Override
    public List<String> tabComplete(CommandSender sender, String alias, String[] args) {
        List<String> options = new ArrayList<>();

        if (args.length == 1) {
            options.add("prelogin");
            options.add("postlogin");
        }

        return options;
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/command/sub/SetCommand.kt
package com.elchologamer.userlogin.command.sub

import com.elchologamer.userlogin.ULPlayer
import com.elchologamer.userlogin.command.base.SubCommand
import com.elchologamer.userlogin.manager.LocationsManager
import com.elchologamer.userlogin.util.QuickMap
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class SetCommand : SubCommand("set", true, "ul.set") {
    override fun run(sender: CommandSender, args: Array<String>): Boolean {
        if (args.isEmpty()) return false

        val type = args[0].lowercase()
        if (type != "spawn" && type != "login") return false

        val player = sender as Player

        // Save location
        val loc = player.location
        LocationsManager.saveLocation(type, loc)

        // Send message
        ULPlayer[player].sendMessage(
            "commands.set",
            QuickMap<String, Any>("type", type)
                .set("x", loc.x.toInt())
                .set("y", loc.y.toInt())
                .set("z", loc.z.toInt())
                .set("yaw", loc.yaw.toInt())
                .set("pitch", loc.pitch.toInt())
                .set("world", loc.world!!.name)
        )
        return true
    }

    override fun tabComplete(sender: CommandSender, label: String, args: Array<String>): List<String> {
        val options: MutableList<String> = ArrayList()

        if (args.size == 1) {
            options.add("login")
            options.add("spawn")
        }

        return options
    }
}